src/libraries/System.Text.Json/gen/Helpers/RoslynExtensions.cs

```diff
@@ -374,6 +374,24 @@ private static bool IsInputTypeNonNullable(this ISymbol symbol, ITypeSymbol inpu
             return inputType.NullableAnnotation is NullableAnnotation.NotAnnotated;
         }
 
+        public static bool IsTuple(this INamedTypeSymbol type)
+        {
+            if (!type.IsGenericType)
+            {
+                return false;
+            }
+
+            string typeFullName = type.ConstructedFrom.ToDisplayString();
```

Comment:
This looks allocatey; consider simply checking if the type is part of mscorlib (by accepting `Compilation` as an additional parameter) and then verifying the name and namespace of the type.